intranet/static/themes/snow/snow.js

```diff
@@ -5,426 +5,427 @@
 var fastfillheight;
 //Graphics control
 var graphics;
 //The multiplier to find the correct bucket
-var heightacc = heightbuckets/screenwidth;
+var heightacc = heightbuckets / screenwidth;
 //Temporary variables
-var newx;
-var bucket;
-var snowsize;
+var newx, bucket, snowsize;
 //Div holding everything, removes scroll bars.
 var container;
 
 var today = new Date(); // what day is it?
 
 //Non-denominational Red Deer-Pulled Guy
-if (today.getMonth() == 11 && today.getDate() <= 25) { // Do not show Red Deer-Pulled Guy after the 25th
-	var santaexists=true; //It's true! I've met him. He's a pretty cool guy.
-}
-var santalink=snowroot+"santa_xsnow.gif";
-var santawidth=210;
-var santaheight=83;
-var santaspeed=5;
-var santax=-santawidth;
+//Do not show Red Deer-Pulled Guy after the 25th
+//It's true! I've met him. He's a pretty cool guy.
+var santaexists = (today.getMonth() === 11 && today.getDate() <= 25);
+var santalink = snowroot + "santa_xsnow.gif";
+var santawidth = 210;
+var santaheight = 83;
+var santaspeed = 5;
+var santax = -santawidth;
 var santa;
 
+function set_urlvars() {
+    var regex = /[\?&]flake=([^&#]*)/;
+    var results = regex.exec(window.location.href);
+
+    if (results !== null)
+        snowletter = url_decode(results[1]);
 
+    regex = /[\?&]colors=([^&#]*)/;
+    results = regex.exec(window.location.href);
 
-function set_urlvars() {
-	var regex = new RegExp("[\\?&]flake=([^&#]*)");
-	var results = regex.exec( window.location.href );
-	if( results != null )
-		snowletter=url_decode(results[1]);
-	var regex = new RegExp("[\\?&]colors=([^&#]*)");
-	var results = regex.exec( window.location.href );
-	if( results != null )
-		snowcolor=extract_color(url_decode(results[1]));
+    if (results !== null)
+        snowcolor = extract_color(url_decode(results[1]));
 }
 
 function url_decode(utftext) {
-	//
-	// Credit for the base for this function
-	// goes to the people at webtoolkit
-	// http://www.webtoolkit.info/
-	// Pretty cool code, imho.
-	//
-	utftext=unescape(utftext);
-	var string = "";
-	var i = 0;
-	var c = c1 = c2 = 0;
-	
-	while ( i < utftext.length ) {
-		c = utftext.charCodeAt(i);
-		if (c < 128) {
-			string += String.fromCharCode(c);
-			i++;
-		} else if((c > 191) && (c < 224)) {
-			c2 = utftext.charCodeAt(i+1);
-			string += String.fromCharCode(((c & 31) << 6) | (c2 & 63));
-			i += 2;
-		} else {
-			c2 = utftext.charCodeAt(i+1);
-			c3 = utftext.charCodeAt(i+2);
-			string += String.fromCharCode(((c & 15) << 12) | ((c2 & 63) << 6) | (c3 & 63));
-			i += 3;
-		}
-	}
-	string = string.replace(/&/g,"&amp;").replace(/</g,"&lt;").replace(/>/g,"&gt;");
-	return string;
+    /*
+     * Credit for the base for this function
+     * goes to the people at webtoolkit
+     * http://www.webtoolkit.info/javascript_url_decode_encode.html
+     * Pretty cool code, imho.
+     */
+    utftext = unescape(utftext);
+    var string = "";
+    var i = 0;
+    var c = 0,
+        c1 = 0,
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['c1' is defined but never used.](https://www.codacy.com/app/tjcsl/ion/file/2746716172/issues/source?bid=3176160&fileBranchId=3355598#l138)